self-run-agents/ami-build/azp-arm-large.json

```diff
@@ -0,0 +1,69 @@
+{
+  "variables": {
+    "azure_pipelines_token": "null"
+  },
+  "builders": [{
+    "type": "amazon-ebs",
+    "region": "us-east-1",
+    "source_ami_filter": {
+      "filters": {
+        "virtualization-type": "hvm",
+        "name": "ubuntu/images/*ubuntu-bionic-18.04-arm64-server-*",
+        "root-device-type": "ebs"
+      },
+      "owners": ["099720109477"],
+      "most_recent": true
+    },
+    "instance_type": "a1.4xlarge",
+    "ssh_username": "ubuntu",
+    "ami_name": "envoy-azp-arm-large-{{timestamp}}",
+    "run_volume_tags": {
+      "Project": "Packer"
+    },
+    "run_tags": {
+      "Project": "Packer"
+    },
+    "tags": {
+      "Project": "envoy-azp-arm-large"
+    },
+    "security_group_ids": [
+      "sg-0e26fd7adddb1b9fa"
+    ],
+    "encrypt_boot": true
+  }],
+  "provisioners": [
+    {
+      "type": "shell",
+      "inline": [
+        "sudo apt-get update",
+        "sudo apt -y install docker.io",
+        "sudo systemctl enable docker",
+        "sudo systemctl start docker"
+      ]
+    },
+    {
+      "type": "shell",
+      "inline": [
+        "sudo useradd -ms /bin/bash -G sudo,docker azure-pipelines",
+        "sudo su root -c -- 'mkdir -p /etc/sudoers.d && echo \"azure-pipelines ALL=(ALL) NOPASSWD:ALL\" | tee /etc/sudoers.d/azure-pipelines && chmod 0440 /etc/sudoers.d/azure-pipelines'"
```

Comment:
What do we need sudo for the agent for?

Reply:
This is to give the exact same experience as azure, Incase some build silently depends on it somewhere.

Reply:
I tend not to because we don't clean up agents when a job finished, in contrast to the hosted agents. If we can cleanup agent (i.e. recreating the VM) we can do this.